python/src/jentic/lib/core_api.py

```diff
@@ -177,9 +177,10 @@ def _validate_response(self, response: Response) -> T_JSONResponse:
         if isinstance(data, list):
             return data
 
-        # If we dont have status code, set it from the response
-        if data.get("output") and data["output"].get("status_code"):
-            data["status_code"] = data["output"]["status_code"]
+        # Check for status code in output, if not set it from the response
+        output = data.get("output")
+        if output and isinstance(output, dict) and output.get("status_code"):
```

Comment:
```suggestion
        if output and isinstance(output, dict) and 'status_code' in output:
```

Do we care what the value of `status_code` key is or we're just copying the value of the key to another location regardless of it's value?

Reply:
yeh we dont care, we just want to pull it out and supply it.  Its not always the same as the response.status_code, as the response.status_code is from our backend, and the output one is from the underlying call.